### ISSUE TITLE
Tooltips don't dismiss on escape

### ISSUE THREAD

**Issue by jdrush89** [primer/react#3921]
### Description

According to https://www.w3.org/TR/WCAG21/#content-on-hover-or-focus content that appears on hover or on focus needs to be dismissible. Pressing the escape key should dismiss tooltips when they are being shown.

### Steps to reproduce

1. Go to https://primer.style/components/tooltip/react
2. Hover over the tooltip in the example
3. Press 'escape'
4. The tooltip is still there

### Version

v36.1.0

### Browser

Chrome

**Comment by joshblack** [comment]
Hi there, @jdrush89! I believe some of the shortcomings in the existing Tooltip component are being addressed with the new component in drafts ([Link](https://primer.style/react/storybook/?path=/story/drafts-components-tooltip--default)).

@broccolinisoup just wanted to check, did I get that right? 👀 And is there any guidance that we have for folks who are dealing with a11y remediations with the existing tooltip?

**Comment by broccolinisoup** [comment]
@joshblack yes, that is perfect!

>And is there any guidance that we have for folks who are dealing with a11y remediations with the existing tooltip?

Not that I am aware of sorry. Might worth to check with #accessibility.

**Comment by lesliecdubs** [comment]
@jdrush89 if you'd like to follow along on further progress on the [new `Tooltip` component](https://primer.style/react/storybook/?path=/story/drafts-components-tooltip--default), you may be interested in subscribing to https://github.com/github/primer/issues/1244. 

We'll also announce in [PRC release notes](https://github.com/primer/react/releases) and the [Primer Changelog](https://github.com/github/primer/discussions/categories/primer-changelog) when the improved `Tooltip` is officially rolling out, if you'd prefer to hold off on adoption until then.

**Comment by broccolinisoup** [comment]
Tooltip v2 is available from the next bundle https://primer.style/components/tooltip/react/beta and adopting it should resolve the problem reported ✨ I'm closing this issue but please feel free to reopen if you have any concern 🙌
